plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -71,6 +92,123 @@ public static boolean hasInvalidStatistics(Collection<ColumnChunkMetaData> metad
+    public static Object toTrinoValue(Type type, @Nullable Object value)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        if (type == SMALLINT || type == TINYINT || type == INTEGER) {
+            return (long) (int) value;
+        }
+        if (type == BIGINT) {
+            return (long) (int) value;
+        }
+        if (type == REAL) {
+            return (long) floatToRawIntBits((float) (double) value);
+        }
+        if (type == DOUBLE) {
+            return value;
+        }
+        if (type instanceof DecimalType) {
+            BigDecimal decimal;
+            checkArgument(value instanceof String || value instanceof Double, "Value must be instance of String or Double");
+            if (value instanceof String) {
+                decimal = new BigDecimal((String) value);
+            }
+            else {
+                decimal = BigDecimal.valueOf((double) value);
+            }
+
+            if (isShortDecimal(type)) {
+                return Decimals.encodeShortScaledValue(decimal, ((DecimalType) type).getScale());
+            }
+            return Decimals.encodeScaledValue(decimal, ((DecimalType) type).getScale());
+        }
+        if (type instanceof VarcharType) {
+            return value;
+        }
+        if (type.equals(DateType.DATE)) {
+            return LocalDate.parse((String) value).toEpochDay();
+        }
+        if (type instanceof TimestampType) {
+            return Instant.parse((String) value).toEpochMilli() * MICROSECONDS_PER_MILLISECOND;
+        }
+        if (type instanceof RowType) {
+            RowType rowType = (RowType) type;
+            Map<?, ?> values = (Map<?, ?>) value;
+            List<Type> fieldTypes = rowType.getTypeParameters();
+            BlockBuilder blockBuilder = new RowBlockBuilder(fieldTypes, null, 1);
+            BlockBuilder singleRowBlockWriter = blockBuilder.beginBlockEntry();
+            for (int i = 0; i < values.size(); ++i) {
+                Type fieldType = fieldTypes.get(i);
+                Object fieldValue = toTrinoValue(fieldType, values.get(rowType.getFields().get(i).getName().orElseThrow()));
+                writeNativeValue(fieldType, singleRowBlockWriter, fieldValue);
+            }
+
+            blockBuilder.closeEntry();
+            return blockBuilder.build();
+        }
+
+        throw new UnsupportedOperationException("Unsupported type: " + type);
+    }
+
+    public static Optional<Map<String, Object>> toJsonValues(Map<String, Type> columnTypeMapping, Optional<Map<String, Object>> values)
+    {
+        if (values.isEmpty()) {
+            return values;
+        }
+
+        Map<String, Object> jsonValues = new HashMap<>();
+        for (Map.Entry<String, Object> value : values.get().entrySet()) {
+            Type type = columnTypeMapping.get(value.getKey());
+            if (type instanceof ArrayType || type instanceof MapType || type instanceof RowType) {
+                continue;
```

Comment:
Should we be converting nested RowTypes?

Reply:
Left a TODO comment, but let me know if the support is mandatory in this PR.